### ISSUE TITLE
rollout() runs num_timesteps+1 transitions after reset (off-by-one vs docstring)

### ISSUE THREAD

**Issue by docxology** [infer-actively/pymdp#429]
## Summary

`pymdp/envs/rollout.py:674` runs `num_timesteps + 1` environment transitions after `reset`, contradicting the documented "for `num_timesteps`" contract and producing one spurious extra action/observation trace.

## Details

```python
# rollout.py:674
last, info = lax.scan(step_fn, initial_carry, jnp.arange(num_timesteps + 1))
```

The docstring (`:477`, `:485`) states it "Roll[s] out ... for `num_timesteps`" / "Number of timesteps to simulate." `reset` already yields `observation_0` (`:648`), and each scan iteration calls `env.step` (`:585`), so the loop emits `num_timesteps + 1` actions and observations `o_0..o_{num_timesteps+1}`.

The decisive internal inconsistency: `action_history_len = max(history_len - 1, 0)` at `:514` with `_resolve_history_len` returning `num_timesteps + 1` (`:45`) sizes the **action** history to exactly `num_timesteps`, while the scan actually produces `num_timesteps + 1` actions. The `+1` inside `_resolve_history_len` is intended padding for the observation buffer (holding `o_t` plus the next `o_{t+1}` context); the `+1` in the scan `arange` is the off-by-one.

## Suggested fix

```python
last, info = lax.scan(step_fn, initial_carry, jnp.arange(num_timesteps))
```

This yields `num_timesteps` actions and `num_timesteps + 1` observations (`o_0..o_{num_timesteps}`), consistent with both the docstring and `action_history_len`. Keep the `+1` in `_resolve_history_len` (it correctly sizes the observation window). Downstream consumers of the observation trace are unaffected in count because observation count stays `num_timesteps + 1` either way.

## Environment

- inferactively-pymdp 1.0.3 (commit 0681ce4)
- Found via static red-team review; cross-checked against `_resolve_history_len` and `action_history_len`
